src/main/java/org/folio/service/transactions/cancel/CancelPaymentCreditService.java

```diff
@@ -1,46 +1,60 @@
 package org.folio.service.transactions.cancel;
 
 import io.vertx.core.json.JsonObject;
+import org.folio.dao.transactions.TransactionDAO;
 import org.folio.rest.jaxrs.model.Budget;
 import org.folio.rest.jaxrs.model.Transaction;
+import org.folio.rest.jaxrs.model.Transaction.TransactionType;
 import org.folio.service.budget.BudgetService;
 import org.folio.utils.MoneyUtils;
 
 import javax.money.CurrencyUnit;
 import javax.money.Monetary;
 import java.util.List;
-import java.util.Map;
-
-import static org.apache.commons.collections4.CollectionUtils.isNotEmpty;
 
 public class CancelPaymentCreditService extends CancelTransactionService {
 
-  public CancelPaymentCreditService(BudgetService budgetService) {
-    super(budgetService);
+  public CancelPaymentCreditService(BudgetService budgetService, TransactionDAO paymentCreditDAO, TransactionDAO encumbranceDAO) {
+    super(budgetService, paymentCreditDAO, encumbranceDAO);
+  }
+
+  @Override
+  Budget budgetMoneyBack(Budget budget, List<Transaction> transactions) {
```

Comment:
Please update visibility methods for overriding in the interface CancelTransactionService and the same for all children classes.

Reply:
All method which should be implemented or overridden in the children better to  make **"protected"**, because you child can be placed in another package and will not be visible in the child.

Reply:
OK